R/db-odbc-postgres.R

```diff
@@ -0,0 +1,61 @@
+#' @export
+sql_translate_env.PostgreSQL <- function(con) {
+
+    # Started with a copy of the existing PostgreSQL translation
+    # and included additional variances that did not pass the
+    # tests from dbtest.  To leverage the rest of the translations,
+    # the bases are set to base_scalar, base_agg and base_win,
+    # instead of base_odbc_...
+
+    sql_variant(
+    sql_translator(.parent = base_scalar,
+       # cot, cosh, sinh, coth and tanh calculations are based on this article
+       # https://en.wikipedia.org/wiki/Hyperbolic_function
+       cot = function(x){
+         build_sql("1 / TAN(", x, ")")
+       },
+      cosh = function(x){
+        build_sql("(EXP(", x, ") + EXP(-", x,")) / 2")
+      },
+      sinh = function(x){
+        build_sql("(EXP(", x, ") - EXP(-", x,")) / 2")
+      },
+      tanh = function(x){
+        build_sql("((EXP(", x, ") - EXP(-", x,")) / 2) / ((EXP(", x, ") + EXP(-", x,")) / 2)")
+      },
+      coth = function(x){
+        build_sql("((EXP(", x, ") + EXP(-", x,")) / 2) / ((EXP(", x, ") - EXP(-", x,")) / 2)")
+      },
+      round = function(x, digits = 0L){
+        build_sql(
+          "ROUND(", x, ", ", as.integer(digits),")"
+        )},
+      log = function(x, base = exp(1)) {
+        if (isTRUE(all.equal(base, exp(1)))) {
+          build_sql("ln(", x, ")")
+        } else {
+          # Use log change-of-base because postgres doesn't support the
+          # two-argument "log(base, x)" for floating point x.
+          build_sql("log(", x, ") / log(", base, ")")
```

Comment:
I used the translation from `db-postgres.r` as the base, so I didn't really question it.  I can double check it.